src/Bicep.Cli/Services/Evaluation.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Deployments.Core.Definitions.Schema;
+
+namespace Bicep.Cli.Services
+{
+    public class Evaluation
+    {
+        public Template? Template { get; }
+        public InvalidOperationException? Error { get; }
+        public Dictionary<string, bool> Assertions { get; }
+        public Dictionary<string, bool> FailedAssertions { get; }
+        public bool Success => Error == null && (FailedAssertions.Count == 0);
+        public bool Skip => Error != null;
+        public Evaluation(Template? template, Dictionary<string, bool>? assertions, InvalidOperationException? error)
+        {
+            Template = template;
+            Assertions = assertions ?? new Dictionary<string, bool>();
+            Error = error;
+            FailedAssertions = assertions?.Where(a => !a.Value).ToDictionary(a => a.Key, a => a.Value) ?? new Dictionary<string, bool>();
```

Comment:
It doesn't feel to me like you actually need this to be a dictionary because you iterate over it as if it were a collection where you actually use it. One big downside to using a dictionary is that you don't have control over the ordering. IMO it would be worth using an `ImmutableArray` and storing the assertions in the order they appear in the Bicep file (to ensure that the order is **always** the same and is consistent with the file). If you need to group related data in a collection, you can always use a `record` or a [tuple type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples) for it.

For example, declaring:
```csharp
record AssertionResult(
    AssertSymbol Source,
    bool Result);
```
Using:
```csharp
public ImmutableArray<AssertionResult> AssertionResults { get; }
```